tools/Config.mk

```diff
@@ -149,10 +149,10 @@ ifeq (,$(wildcard $(CUSTOM_BOARD_KPATH)))
 else
   BOARD_KCONFIG = $(CUSTOM_BOARD_KPATH)
 endif
-
-BOARD_COMMON_DIR ?= $(wildcard $(BOARD_DIR)$(DELIM)..$(DELIM)common)
-ifeq ($(BOARD_COMMON_DIR),)
-  BOARD_COMMON_DIR = $(wildcard $(TOPDIR)$(DELIM)boards$(DELIM)$(CONFIG_ARCH)$(DELIM)$(CONFIG_ARCH_CHIP)$(DELIM)common)
+ifeq ($(CONFIG_BOARD_CUSTOM_ARCH_BOARD_COMMON),y)
+  BOARD_COMMON_DIR ?= $(wildcard $(TOPDIR)$(DELIM)boards$(DELIM)$(CONFIG_ARCH)$(DELIM)$(CONFIG_ARCH_CHIP)$(DELIM)common)
```

Comment:
BOARD_COMMON_DIR is used inside Make.defs, if user don't like the built-in common board, he can ignore this variable directly, I think.